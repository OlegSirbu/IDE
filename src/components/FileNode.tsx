import React from 'react';

const FileNode: React.FC<FileNode> = ({
  name,
  isSelected,
  fullPath,
  onSelect
}) => {
  return (
    <div
      onClick={() => onSelect(fullPath)}
      className={`pointer ${isSelected ? 'selected-file' : ''}`}>
      ðŸ“„ {name}
    </div>
  );
};

export default FileNode;
