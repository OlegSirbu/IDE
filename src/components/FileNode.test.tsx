import React from 'react';
import { render } from '@testing-library/react';
import FileNode from './FileNode';

describe('FileNode', () => {
  it('renders correctly and matches snapshot', () => {
    const { asFragment } = render(
      <FileNode
        name="exampleFile.txt"
        isSelected={false}
        fullPath="/path/to/exampleFile.txt"
        onSelect={() => {}}
      />
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('renders correctly when selected and matches snapshot', () => {
    const { asFragment } = render(
      <FileNode
        name="exampleFile.txt"
        isSelected={true}
        fullPath="/path/to/exampleFile.txt"
        onSelect={() => {}}
      />
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
