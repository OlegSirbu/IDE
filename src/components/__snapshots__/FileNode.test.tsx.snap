// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FileNode renders correctly and matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="pointer "
  >
    ðŸ“„ exampleFile.txt
  </div>
</DocumentFragment>
`;

exports[`FileNode renders correctly when selected and matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="pointer selected-file"
  >
    ðŸ“„ exampleFile.txt
  </div>
</DocumentFragment>
`;
